tests/BiosTests.hs

```diff
@@ -190,30 +202,32 @@ testLoadFile crd a_fp step = do
 
 testLoadFileCradleFail :: Cradle a -> FilePath -> (CradleError -> Expectation) -> (String -> IO ()) -> IO ()
 testLoadFileCradleFail crd a_fp cradleErrorExpectation step = do
-  libDir <- getRuntimeGhcLibDir crd
-  withCurrentDirectory (cradleRootDir crd) $
-    G.runGhc libDir $ do
-      let relFp = makeRelative (cradleRootDir crd) a_fp
-      res <- initializeFlagsWithCradleWithMessage (Just (\_ n _ _ -> step (show n))) relFp crd
-      case res of
-        CradleSuccess _ -> liftIO $ expectationFailure "Cradle loaded successfully"
-        CradleNone -> liftIO $ expectationFailure "Unexpected none-Cradle"
-        CradleFail crdlFail -> liftIO $ cradleErrorExpectation crdlFail
+  -- don't spin up a ghc session, just run the opts program manually since
+  -- we're not guaranteed to be able to get the ghc libdir if the cradle is
+  -- failing
```

Comment:
That is interesting and not documented: https://github.com/mpickering/hie-bios/blob/master/src/HIE/Bios/Types.hs#L60
Or what do you mean with this?

Reply:
Now that hie-bios no longer uses ghc-paths it has to call `cabal exec ghc` for the failing-cabal test case to set up the ghc session. But `cabal exec ghc` fails because it can’t resolve the constraints in the first place! So we aren’t able to get a lib dir. Prior to this it was just working by fluke, since there’s no cabal.project in this directory `cabal exec ghc` was using the hie-bios project to resolve it fine. 
We can still test loading the flags etc and it will still fail, we just need to do it without a ghc session

Reply:
Maybe we should document it better that we require `runCradle` to succeed and how the results for libdir may be non-sensical if it fails?